demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.test.alloy.showcase.datatable;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.interactions.Action;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import com.liferay.faces.test.alloy.showcase.data.DataTester;
+import com.liferay.faces.test.selenium.Browser;
+import com.liferay.faces.test.selenium.assertion.SeleniumAssert;
+
+
+/**
+ * @author  Kyle Stiemann
+ * @author  Philip White
+ */
+public class DataTableTester extends DataTester {
+
+	// Protected Constants
+	protected static final String[] defaultCUSTOMERS = {
+			"John Adams", "Samuel Adams", "Josiah Bartlett", "Carter Braxton", "Charles Carroll", "Benjamin Franklin",
+			"Elbridge Gerry", "John Hancock", "Thomas Jefferson", "Francis Lewis"
+		};
+	protected static final String[] firstNameSortCUSTOMERS = {
+			"Adoniram Taverner", "Andrew Colman", "Andrew Kingston", "Andrew Peale", "Anthony Gilby", "Anthony Langley",
+			"Arthur Middleton", "Arthur Roborough", "Barnaby Benison", "Benjamin Franklin"
+		};
+
+	// Common Xpath
+	protected static final String firstNameHeaderxpath = "//table/thead/tr/th[@scope='col'][contains(.,'First Name')]";
+	protected static final String lastNameHeaderxpath = "//table/thead/tr/th[@scope='col'][contains(.,'Last Name')]";
+	protected static final String dateOfBirthHeaderxpath =
+		"//table/thead/tr/th[@scope='col'][contains(.,'Date of Birth')]/div";
+
+	protected String alloyMessageXpath(String alloyMessage) {
+		return "(//table[contains(@class,'alloy-messages')]/tbody/tr/td[contains(.,'" + alloyMessage + "')])";
+	}
+
+	/**
+	 * Returns a string representation of the contents of the specified array. it also replaces any front and back
+	 * square brackets with a value of null.
+	 */
+	protected String arrayToString(String[] stringArray) {
+		return Arrays.toString(stringArray).replaceAll("^\\[", "").replaceAll("\\]$", "");
+	}
+
+	/**
+	 * Must build this action using keyDown(key) and keyUp(key) since Action sendKeys behaves differently than
+	 * WebElement sendKeys (unless the browser is phantomjs, then it would need the following script on line 81).
+	 * https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/interactions/Actions.html#sendKeys-java.lang.CharSequence...-
+	 */
+	protected void controlOrCommandClick(Browser browser, String xpath) {
```

Comment:
I wrongly told you that the `[Ctrl]` was the correct key for non-OSX systems. Please change this method to use the `Keys.META` key (and change the method name to `metaOrCommandClick()`).